retroshare-gui/src/gui/TheWire/PulseAddDialog.cpp

```diff
@@ -510,12 +521,29 @@ void PulseAddDialog::toggle()
 {
 	if (ui.pushButton_picture->isChecked())
 	{
+        // Show the input methods (drag and drop field and the browse button)
 		ui.frame_picture->show();
+        ui.frame_PictureBrowse->show();
+
 		ui.pushButton_picture->setToolTip(tr("Hide Pictures"));
 	}
 	else
 	{
+        // Hide the input methods (drag and drop field and the browse button)
 		ui.frame_picture->hide();
+        ui.frame_PictureBrowse->hide();
+
 		ui.pushButton_picture->setToolTip(tr("Add Pictures"));
 	}
 }
+
+// Function to get the file dialog for the browse button
+void PulseAddDialog::onBrowseButtonClicked()
+{
+    QString filePath = QFileDialog::getOpenFileName(this, tr("Select image file"), QString(), tr("Image files (*.png *.jpg *.jpeg *.bmp *.gif)"));
```

Comment:
it's better to use misc::getOpenFilename(), as it taks care of error handling (Qt::DontUseNativeDialog), keeps the last directory used, and centralizes all calls to this Qt function.

Reply:
hello, i made the changes please review them. Thanks.